packages/web-components/src/wallet-connection/watchWallet.js

```diff
@@ -65,33 +64,28 @@ export const watchWallet = async (chainStorageWatcher, address) => {
     ),
   );
 
-  // NB: this watches '.current' but only notifies of changes to offerToPublicSubscriberPaths
-  await /** @type {Promise<void>} */ (
-    new Promise((res, rej) => {
-      let lastPaths;
-      chainStorageWatcher.watchLatest(
-        ['data', `published.wallet.${address}.current`],
-        value => {
-          res();
-          const { offerToPublicSubscriberPaths: currentPaths } = value;
-          if (currentPaths === lastPaths) return;
-
-          publicSubscriberPathsNotifierKit.updater.updateState(
-            harden(currentPaths),
-          );
-        },
-        err => {
-          if (!lastPaths) {
-            rej();
-          } else {
-            throw Error(err);
-          }
-        },
+  const smartWalletStatusNotifierKit = {};
+
+  let lastPaths;
+  chainStorageWatcher.watchLatest(
+    ['data', `published.wallet.${address}.current`],
+    value => {
+      smartWalletStatusNotifierKit.provisioned = true;
+      const { offerToPublicSubscriberPaths: currentPaths } = value;
+      if (currentPaths === lastPaths) return;
+
+      publicSubscriberPathsNotifierKit.updater.updateState(
+        harden(currentPaths),
       );
-    })
-  ).catch(() => {
-    throw Error(Errors.noSmartWallet);
-  });
+    },
+    err => {
+      if (!lastPaths) {
+        smartWalletStatusNotifierKit.provisioned = false;
```

Comment:
I just realized this would falsely report no smart wallet even if it's just that the RPC node is down. This is an existing issue though, so not necessary to fix in this PR, just making note.